### PR TITLE
Storage: Blob.exists() does not work within Batch context

### PR DESCRIPTION
**Pull request by IlyaFaer**
My proposition is to add something like future result - boolean in our case. An object will be working as original `bool`, but under the hood it'll be checking if future response is done without exceptions

Towards: #5388

### CODE REVIEW

File: storage/google/cloud/storage/blob.py [hunk 1]
```diff
@@ -2057,3 +2064,29 @@ def _add_query_parameters(base_url, name_value_pairs):
...
+
+    This can be used as a target object for the future request
+    (which is part of batch mechanism) to get result of this request.
+    """
+    _properties = {}
+
+    @property
+    def _is_successfull(self):
+        """Checks if binded future request have been completed without any exception."""
+        return not isinstance(self._properties, _FutureDict)
```
Comment (IlyaFaer) [inline storage/google/cloud/storage/blob.py]: If exception have been raise or request is not done yet, there will be `_FutureDict` obj

File: storage/tests/unit/test_blob.py [hunk 2]
```diff
@@ -20,13 +20,45 @@
...
+        content=b"""
+--batch_pK7JBAk73-E=_AA5eFwv4m2Q=
+HTTP/1.1 404 NotFound
+
+--batch_pK7JBAk73-E=_AA5eFwv4m2Q=
+HTTP/1.1 200 OK
+
+--batch_pK7JBAk73-E=_AA5eFwv4m2Q=--"""
+    )
+    return mock_conn
```
Comment (IlyaFaer) [inline storage/tests/unit/test_blob.py]: Here are two different responses for two different `exists()` calls, and one of them is NotFound

File: storage/tests/unit/test_blob.py [hunk 3]
```diff
@@ -582,6 +614,27 @@ def test_exists_miss(self):
...
+
+        try:
+            with client.batch():
+                bool1 = blob1.exists()
+                bool2 = blob2.exists()
+        except NotFound:
+            pass
+
+        self.assertFalse(bool1)
+        self.assertTrue(bool2)
```
Comment (IlyaFaer) [inline storage/tests/unit/test_blob.py]: Checking if both `exists()` calls working as they suppose to: the first one must be `False` (NotFound)

**Comment by mf2199** [comment]
@tseaver  @crwilcox  @frankyn A friendly reminder: This PR has been hanging out for awhile. We could use some feedback regarding its status.

**Comment by frankyn** [comment]
Hi @IlyaFaer,

I'm closing this PR as it's focused on a feature request that hasn't been prioritized or has a reviewed design.

Please prioritize bugs in the client before feature requests.

Thank you for your patience.

**Pull request closed without merging**